### PR TITLE
Add mozsocial waitlist page [fix #13635]

### LINKED ISSUE #13635: Create a waiting list signup page for mozilla.social
## Description

The Mozilla Social team wants to move their waiting list from the current system to use our standard bedrock -> basket -> CTMS -> Acoustic one. We will need to create the form on bedrock and host it at a URL they've listed in the doc linked below (or one we feel is better if we want to pitch them something else).

[Spec doc](https://docs.google.com/document/d/1lczzwGgkNi50IiH3hlaR6HjNYnzw2PKHo3tr2Tdicdk/edit#heading=h.lxk671tolfxh)

The [existing form is here](https://prod.oidc-proxy.prod.webservices.mozgcp.net/waitlist).

The extra non-standard fields will pass through basket to CTMS as long as they're named appropriately with the waiting list's prefix, much like the country and platform fields on other waiting list forms. 

For example:

The slug for this waiting list is `mozilla-social-waitlist`, so the extra fields should be something like:

* `mozilla_social_twitter_handle`
* `mozilla_social_mastodon_handle`

In other words, you remove `-waitlist` from the name, replace the `-` with `_`, and add a unique field name after.

### PR DESCRIPTION
**Pull request by craigcook**
## One-line summary
Adds a waitlist signup page for mozilla.social.

## Issue / Bugzilla link
#13635 

## Testing
http://localhost:8000/products/mozsocial/invite/

### CODE REVIEW

**Review by nathan-barrett — APPROVED** [review]
Nice work getting this wrapped up @craigcook, here is a **r+** for you

**Comment by craigcook** [comment]
Just waiting on verification from stakeholders, it's possible we'll need to add a hidden custom field so they can sort things in Acoustic, unless they can find an easy method on the back-end without it. So we'll hold off merging for the moment until I get an official go-ahead.

**Comment by craigcook** [comment]
We got the green light 🟢

**Pull request merged**
